ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -0,0 +1,168 @@
+  private final DiskLevelMetrics metrics;
+
+  private static final Logger logger = LoggerFactory.getLogger(DiskManager.class);
+
+  /**
+   * Constructs a {@link DiskManager}
+   * @param disk representation of the disk.
+   * @param replicas all the replicas on this disk.
+   * @param config the settings for store configuration.
+   * @param scheduler the {@link ScheduledExecutorService} for executing background tasks.
+   * @param storeMetrics the {@link StoreMetrics} object used for store-related metrics.
+   * @param keyFactory the {@link StoreKeyFactory} for parsing store keys.
+   * @param recovery the {@link MessageStoreRecovery} instance to use.
+   * @param hardDelete the {@link MessageStoreHardDelete} instance to use.
+   * @param time the {@link Time} instance to use.
+   */
+  DiskManager(DiskId disk, List<ReplicaId> replicas, StoreConfig config, ScheduledExecutorService scheduler,
+      StoreMetrics storeMetrics, StoreKeyFactory keyFactory, MessageStoreRecovery recovery,
+      MessageStoreHardDelete hardDelete, Time time) {
+    this.disk = disk;
+    this.config = config;
+    this.metrics = storeMetrics.createDiskLevelMetrics(disk.getMountPath());
+    DiskIOScheduler ioScheduler = new DiskIOScheduler(null);
+    for (ReplicaId replica : replicas) {
+      if (disk.equals(replica.getDiskId())) {
+        String storeId = replica.getPartitionId().toString();
+        BlobStore store = new BlobStore(storeId, config, scheduler, ioScheduler, metrics, replica.getReplicaPath(),
+            replica.getCapacityInBytes(), keyFactory, recovery, hardDelete, time);
+        stores.put(replica.getPartitionId(), store);
+      }
+    }
+  }
+
+  /**
+   * Starts all the stores on this disk.
+   */
+  void start()
+      throws InterruptedException {
+    Timer.Context context = metrics.diskStartTime.time();
+    try {
+      File mountPath = new File(disk.getMountPath());
+      if (mountPath.exists()) {
+        Map<StoreStarter, Thread> startupThreads = new HashMap<>();
+        for (final Map.Entry<PartitionId, BlobStore> partitionAndStore : stores.entrySet()) {
+          StoreStarter storeStarter = new StoreStarter(partitionAndStore.getKey(), partitionAndStore.getValue());
+          Thread thread = Utils.newThread("store-startup-" + partitionAndStore.getKey(), storeStarter, false);
+          thread.start();
+          startupThreads.put(storeStarter, thread);
+        }
+        int numFailed = 0;
+        for (Map.Entry<StoreStarter, Thread> starterAndThread : startupThreads.entrySet()) {
+          starterAndThread.getValue().join();
+          Exception exception = starterAndThread.getKey().exception;
+          if (exception != null) {
+            numFailed++;
+            logger.error("Exception while starting store for the partition" + starterAndThread.getKey().partition,
+                exception);
+          }
+        }
+        if (numFailed > 0) {
+          metrics.diskStartFailure.inc();
```

Comment:
it seems like you might want to keep this metric different from the metric for non-existent mount path (a.k.a disk failure).